interfaces.go

```diff
@@ -79,6 +79,9 @@ type Method interface {
 
 	// Stores the next location to evaluate in xNext
 	Iterate(loc *Location, xNext []float64) (EvaluationType, IterationType, error)
+
+	// Requires returns the minimum requirements of Method on Function.
```

Comment:
I mean, `Requires()` is just a summary of what `Init()` and `Iterate()` return during the run. The matching of EvaluationTypes to what Function implements takes place even if `Requires()` does not exist, so we should clearly document it anyway. I think `Function` will be the best place for it since anyone calling `Local()` will look it up in order to implement their objective function.